### PR TITLE
Make `eval` an explicit subcommand and restructure the CLI somewhat

### PR DESCRIPTION
**Pull request by vkleen**
This change makes `eval` an explicit subcommand for the Nickel CLI and adds the ability to pass multiple files to a single CLI invocation. If multiple files are parsed, they will be evaluated separately and the results merged.

Additionally, this PR decouples the customize mode from `nickel export` and instead makes it available anywhere full evaluation is performed.

### CODE REVIEW

File: cli/src/customize.rs [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+const EXPERIMENTAL_MSG: &str =
+    "[WARNING] Customize mode is experimental. Its interface is subject to breaking changes.";
+
+#[derive(clap::Parser, Debug)]
+pub struct CustomizeMode {
+    /// \[WARNING\] Customize mode is experimental. Its interface is subject to breaking changes.
+    ///
+    /// Customize mode turns the nickel invocation into a new CLI based on the configuration to be
+    /// exported, where the value of fields can be set directly through CLI arguments. Print the
+    /// corresponding help (`nickel export -f <file.ncl> -- --help`) to see available options.
```
Comment (jneem) [inline cli/src/customize.rs:39]:
```suggestion
    /// corresponding help (`nickel export <file.ncl> -- --help`) to see available options.
```

File: cli/src/input.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+use std::path::PathBuf;
+
+use nickel_lang_core::{eval::cache::lazy::CBNCache, program::Program};
+
+use crate::{cli::GlobalOptions, customize::Customize, error::CliResult};
+
+#[derive(clap::Parser, Debug)]
+pub struct InputOptions<Customize: clap::Args> {
+    /// Input file, omit to read from stdin
```
Comment (jneem) [inline cli/src/input.rs:9]:
```suggestion
    /// Input files, omit to read from stdin
```

File: cli/src/export.rs [hunk 3]
```diff
@@ -1,44 +1,19 @@
...
-/// The value name used through the CLI to indicate that an option accepts any Nickel expression as
-/// a value.
-const NICKEL_VALUE_NAME: &str = "NICKEL EXPRESSION";
-
-const EXPERIMENTAL_MSG: &str =
-    "[WARNING] Customize mode is experimental. Its interface is subject to breaking changes.";
-
 #[derive(clap::Parser, Debug)]
 pub struct ExportCommand {
     #[arg(long, value_enum, default_value_t)]
```
Comment (jneem) [inline cli/src/export.rs:19]: I guess this can get a short option now, right?

**Review by jneem — APPROVED** [review]

File: cli/src/export.rs (hunk 3 above)
Comment (vkleen) [reply cli/src/export.rs:19]: Oh right, yes it can.

File: cli/src/eval.rs [hunk 4]
```diff
@@ -2,39 +2,44 @@ use nickel_lang_core::{eval::cache::lazy::CBNCache, program::Program};
...
+    customize::CustomizeMode,
     error::{CliResult, ResultErrorExt},
+    input::{InputOptions, Prepare},
 };
 
 #[derive(clap::Parser, Debug)]
-pub struct EvalCommand {}
+pub struct EvalCommand {
+    #[cfg(debug_assertions)]
+    /// Skips the standard library import. For debugging only. This does not affect REPL
```
Comment (yannham) [inline cli/src/eval.rs:13]: Now that this has become a flag of `eval` only, I guess you can't even use it with the REPL anymore?

```suggestion
    /// Skips the standard library import. For debugging only
```

**Review by yannham — APPROVED** [review]

**Pull request merged**